packages/sandbox/src/file_watching_sandbox.test.ts

```diff
@@ -854,18 +897,23 @@ const setupAndStartSandbox = async (
     testData.open ?? _open
   );
 
-  await sandboxInstance.start({
-    dir: testData.dir,
-    exclude: testData.exclude,
-    identifier: testData.name,
-    format: testData.format,
-    profile: testData.profile,
-  });
+  await sandboxInstance.start(sandboxOptions);
 
   // At this point one deployment should already have been done on sandbox startup
   assert.strictEqual(backendDeployerDeployMock.mock.callCount(), 1);
   // and client config generated only once
 
+  if (resetMocksAfterStart) {
+    // Reset all the calls to avoid extra startup call
+    backendDeployerDestroyMock.mock.resetCalls();
+    backendDeployerDeployMock.mock.resetCalls();
+    cfnClientSendMock.mock.resetCalls();
+    listSecretMock.mock.resetCalls();
+  }
+
+  if (!sandboxOptions.watchForChanges) {
+    return { sandboxInstance, fileChangeEventCallback: () => null };
```

Comment:
nit: it should throw, something like
```suggestion
    return { sandboxInstance, fileChangeEventCallback: () => throw new Error('When not watching for changes, file change event callback is not available' };
```